### PR TITLE
[WIP] adding https guide to installation guide

### PR DESCRIPTION
**Pull request by alex9311**
will someone please try out the steps in this guide

#452

### CODE REVIEW

File: cvat/apps/documentation/installation.md [hunk 1]
```diff
@@ -308,3 +308,53 @@ volumes:
...
+
+In order to take `docker-compose.override.yml` into account when starting up CVAT, reference the overriding file with
+`-f` in the `docker-compose up` command.
+Also, don't forget to `docker-compose build` after making your changes.
+```bash
+docker-compose build
+docker-compose -f docker-compose.yml -f docker-compose.override.yml up -d
+```
+
+Those changes should get CVAT running over SSH.
```
Comment (bsekachev) [inline cvat/apps/documentation/installation.md]: SSH?

File: cvat/apps/documentation/installation.md [hunk 2]
```diff
@@ -308,3 +308,53 @@ volumes:
 You can change the share device path to your actual share. For user convenience
 we have defined the environment variable $CVAT_SHARE_URL. This variable
 contains a text (url for example) which is shown in the client-share browser.
+
+### Running over HTTPS
+
+First, you'll want to set the `SECURE_SSL_REDIRECT` inside your `cvat/settings/base.py` file.
+This tells Django that you want to use SSL.
+```python
+SECURE_SSL_REDIRECT = True
```
Comment (bsekachev) [inline cvat/apps/documentation/installation.md]: It seems to me, we should do it through environment variables

Comment (bsekachev) [reply cvat/apps/documentation/installation.md]: Instead of the hardcoded True value

**Comment by nmanovic** [comment]
@alex9311 , is it possible to provide instructions which don't require modify source code? It is OK to add a couple of more files or redefine some environment variables but I don't think that it is good to ask users to modify original source code to enable a feature.

**Comment by nmanovic** [comment]
@alex9311 , one more comment. Need to add information about the feature into CHANGELOG.md

**Comment by nmanovic** [comment]
@alex9311 , will you find time to fix our comments and slightly improve the PR?

**Comment by alex9311** [comment]
hi @nmanovic , yeah I should have time this weekend

**Comment by alex9311** [comment]
I can easily read from an environment variable to set `SECURE_SSL_REDIRECT` instead of asking the user to modify `cvat/settings/base.py`. Are you also unhappy with the use of a `docker-compose.override.yml` file? Or is that OK?

The generation of SLL certificates in the Dockerfile is just a suggestion, there are a lot of different ways users could get those key and cert files. I could remove the Docker suggestion and just tell the user they will need to put their certs at `/etc/ssl/cvat-certs/`.

**Comment by nmanovic** [comment]
@alex9311 ,

> I can easily read from an environment variable to set SECURE_SSL_REDIRECT instead of asking the user to modify cvat/settings/base.py.

It will be better because in this case we can use docker-compose.override.yml to provide the value.

> Are you also unhappy with the use of a docker-compose.override.yml file? Or is that OK?

It is OK.

> The generation of SLL certificates in the Dockerfile is just a suggestion, there are a lot of different ways users could get those key and cert files. I could remove the Docker suggestion and just tell the user they will need to put their certs at /etc/ssl/cvat-certs/.

Probably it is better to ask users to put certificates into a directory on host (e.g. `cvat source code`/ssl). But if you give an example how to generate a dummy certificate in your README file it will be good.

File: cvat/apps/documentation/installation.md (hunk 1 above)
Comment (thusinh1969) [reply cvat/apps/documentation/installation.md]: Would really LOVE to know how to setup SSL. Keys can be getting from Cloudflare for examples ?

Thanks much.
Steve

**Comment by nmanovic** [comment]
@alex9311 , please let me know if you are going to fix comments.

**Comment by nmanovic** [comment]
@alex9311 , it seems you don't have time to finish the PR. I will close it right now.

**Pull request closed without merging**